src/popup/router/pages/TransactionDetails.vue

```diff
@@ -201,26 +228,74 @@ export default {
       'getTxTipUrl',
       'getExplorerPath',
       'isTxAex9',
+      'getDexContracts',
     ]),
     amount() {
       return this.getTxAmountTotal(this.transaction);
     },
     symbol() {
       return this.getTxSymbol(this.transaction);
     },
-    direction() {
-      return this.getTxDirection(this.transaction);
-    },
     tipUrl() {
       return this.getTxTipUrl(this.transaction);
     },
+    singleToken() {
+      return {
+        isReceived: this.getTxDirection(this.transaction) === 'received',
+        amount: this.getTxAmountTotal(this.transaction),
+        symbol: this.getTxSymbol(this.transaction),
+        isAex9: this.isTxAex9(this.transaction),
+        preventAmountConversion: true,
+        showFiat: true,
+      };
+    },
+    isDex() {
+      return this.getDexContracts.router.includes(this.transaction?.tx?.contractId)
+        || this.getDexContracts.wae.includes(this.transaction?.tx?.contractId)
+        || this.isAllowance;
+    },
+    isSwap() {
+      return FUNCTION_TYPE_DEX.swap.includes(this.transaction?.tx?.function);
+    },
+    isPool() {
+      return FUNCTION_TYPE_DEX.pool.includes(this.transaction?.tx?.function);
+    },
+    isAllowance() {
+      return FUNCTION_TYPE_DEX.allowance.includes(this.transaction?.tx?.function);
+    },
+    hasError() {
+      return this.transaction?.tx.result === 'abort';
+    },
   },
   async mounted() {
     this.transaction = this.getTx(this.hash);
     if (!this.transaction || this.transaction?.incomplete) {
       await this.$watchUntilTruly(() => this.$store.state.middleware);
+      await this.$watchUntilTruly(() => this.$store.state.sdk);
       this.transaction = await this.$store.state.middleware.getTxByHash(this.hash);
     }
+    Promise.all(this.tokens().map(
+      async (token) => ({ ...token, tokens: await this.prepareTokenPair(token) }),
+    )).then((values) => {
+      this.tokenList = values.filter((token) => !Number.isNaN(Number(token.amount)));
+    });
+  },
+  methods: {
+    ...mapActions('fungibleTokens', ['getContractTokenPairs']),
+    aettosToAe,
+    async prepareTokenPair(token) {
```

Comment:
All the needed information is in the resolvers output, no need to use getContractTokenPairs.